### PR TITLE
ceph-setup: move the git diff check before calling autogen.sh

### PR DESCRIPTION
**Pull request by alfredodeza**
Signed-off-by: Alfredo Deza <adeza@redhat.com>

### CODE REVIEW

**Comment by ktdreyer** [comment]
LGTM, and it sounds like we can revert after Hammer is EOL

**Comment by smithfarm** [comment]
@alfredodeza Can you add a `Fixes: http://tracker.ceph.com/issues/17901` line before Signed-off-by ?

**Comment by dmick** [comment]
@ktdreyer don't see any need to revert; it was always a risk for no good reason.  If anything, I'd put a commit into hammer with the newer m4 file(s).

**Comment by smithfarm** [comment]
I guess this isn't a problem in jewel because `src/gmock` is a submodule there?

File: ceph-setup/build/build [hunk 1]
```diff
@@ -53,6 +53,25 @@ else
     export CEPH_EXTRA_CMAKE_ARGS="$CEPH_EXTRA_CMAKE_ARGS -DALLOCATOR=tcmalloc"
 fi
 
+
+# When using autotools/autoconf it is possible to see output from `git diff`
```
Comment (dmick) [inline ceph-setup/build/build]: if we're gonna add three lines of comment (great!) let's actually call specific attention to "autogen.sh calls aclocal with --install, which can..." for documentation/archaeologists' sakes

**Comment by dmick** [comment]
a tweak from me and one from @smithfarm and it's ready to go afaic

**Comment by ktdreyer** [comment]
LGTM

**Comment by smithfarm** [comment]
Thanks again, everyone.

**Pull request merged**